assets/js/translator.js

```diff
@@ -743,11 +743,32 @@ function translateDoc() {
                         downloadBrowserWarn();
                         $('div#fileUploadProgress').parent().fadeOut('fast');
                         $('div#fileLoading').fadeOut('fast', function () {
-                            var URL = window.URL || window.webkitURL;
-                            $('a#fileDownload')
-                                .attr('href', URL.createObjectURL(xhr.response))
-                                .attr('download', fileName)
-                                .fadeIn('fast');
+                            if(typeof window.navigator.msSaveBlob !== 'undefined') {
+                                $('a#fileDownload')
+                                    .click(function () {
+                                        window.navigator.msSaveBlob(xhr.response, fileName);
+                                    })
+                                    .fadeIn('fast');
+                            }
+                            else if(/^((?!chrome|android).)*safari/i.test(window.navigator.userAgent)) {
+                                var reader = new FileReader();
+                                reader.onload = function (e) {
+                                    $('a#fileDownload')
+                                        .click(function () {
+                                            window.location.href = reader.result.replace(/^data:[^;]*;/, 'data:attachment/file;');
+                                        })
+                                        .fadeIn('fast');
+                                        $('#fileDownloadBrowserWarning').show();
+                                }
+                                reader.readAsDataURL(xhr.response);
+                            }
+                            else {
+                                var URL = window.URL || window.webkitURL;
+                                $('a#fileDownload')
+                                    .attr('href', URL.createObjectURL(xhr.response))
+                                    .attr('download', fileName)
+                                    .fadeIn('fast');
```

Comment:
Looks like we're always `.fadeIn('fast')`ing? Can we move that out of the conditional block?